Fix build with old clang compilers, e.g. on OS X 10.8 Mountain Lion
https://jira.mongodb.org/browse/SERVER-31016
--- src/mongo/db/operation_context.cpp.orig	2017-10-18 11:01:51.000000000 -0500
+++ src/mongo/db/operation_context.cpp	2017-10-31 16:31:48.000000000 -0500
@@ -325,7 +325,7 @@
         deadline = std::min(deadline, getDeadline());
     }
 
-    const auto waitStatus = [&] {
+    const auto waitStatus = [&]() -> stdx::cv_status {
         if (Date_t::max() == deadline) {
             cv.wait(m);
             return stdx::cv_status::no_timeout;
--- src/mongo/db/operation_context_test.cpp.orig	2017-10-18 11:01:51.000000000 -0500
+++ src/mongo/db/operation_context_test.cpp	2017-10-31 16:31:48.000000000 -0500
@@ -231,7 +231,7 @@
                                                                  Date_t maxTime) {
 
         auto barrier = std::make_shared<unittest::Barrier>(2);
-        auto task = stdx::packaged_task<stdx::cv_status()>([=] {
+        auto task = stdx::packaged_task<stdx::cv_status()>([=]() -> stdx::cv_status {
             if (maxTime < Date_t::max()) {
                 txn->setDeadlineByDate(maxTime);
             }
